### ISSUE TITLE
--version in v3.1.2 gives wrong version when building from source

### ISSUE THREAD

**Issue by agro1986** [prysmaticlabs/prysm#11782]
# 🐞 Bug Report

### Description

```
git checkout v3.1.2
~/go/bin/bazelisk build //cmd/beacon-chain:beacon-chain --config=release
~/go/bin/bazelisk build //cmd/validator:validator --config=release
./bazel-bin/cmd/beacon-chain/beacon-chain_/beacon-chain --version
# beacon-chain version Prysm/v3.2.0-rc.0/648ab9f2c249f1d06d0aad4328e8df429ceaf66c. Built at: xxx
./bazel-bin/cmd/validator/validator_/validator --version
# validator version Prysm/v3.2.0-rc.0/648ab9f2c249f1d06d0aad4328e8df429ceaf66c. Built at: xxx
```

It should output v3.1.2 as the version (when fixed in next version, should display v3.1.3, v3.1.4, etc)

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
Ubuntu Server 22.04.1
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
v3.1.2
  </code>
</pre>

**Comment by prestonvanloon** [comment]
Interesting... The script seems to pull the latest tag from git instead of the tag of the commit you are actually on.

**Comment by prestonvanloon** [comment]
https://github.com/prysmaticlabs/prysm/blob/e843cafe7d604e9f80937797d31a7c0d46f60070/hack/workspace_status.sh#L9

This line is incorrect. Perhaps it should be `git describe --exact-match --tags $(git log -n1 --pretty='%h')`, but that doesn't work when you are building from a commit that doesn't have a tag. I'm not sure if there is a nicer way to query git to say "what is the most recent tag at or before the commit that I am currently on"

**Comment by dyng** [comment]
@prestonvanloon Is `git describe --tags --abbrev=0` what you want? It prints most recent tag dates back from `HEAD`.

**Comment by prestonvanloon** [comment]
@dyng we want STABLE_GIT_TAG to resolve to the tag that is present on the current commit, or the latest tag before the current commit. Right now, it is resolving to a tag that is placed after the current commit.

In original issue, they tried to build from the commit tagged as v3.1.2 and STABLE_GIT_TAG resolved to the most recent tag of v3.2.0-rc.0.
